### DISCUSSION TITLE
dm3/dm4 to tif

### DISCUSSION THREAD

**Discussion by apendersG** [hyperspy/hyperspy#3473] [Q&A]
I'm struggling converting a dm3/dm4 file to tif file format. The problem relies in the fact that s.data.dtype='dtype('float32')' and that it does not scale well when converting to another format. I've tried rescaling the data to np.float16 but this does not stretch the pixel histogram well. Additional stretching of the data worked, but felt too complicated plus introduced other artefacts.

`s=hs.load('O.dm3')
s.save('O.tif')`

> s.data
Out  [2]: 
array([[509.02533, 516.7191 , 505.1758 , ..., 440.4813 , 423.06055,
        452.32156],
       [533.15063, 482.31668, 529.44836, ..., 457.14758, 426.31845,
        439.16064],
       [511.56256, 481.27292, 516.69543, ..., 467.83334, 455.2274 ,
        430.6732 ],
       ...,
       [578.1487 , 597.9811 , 551.2473 , ..., 174.62825, 165.9534 ,
        176.75775],
       [526.0436 , 541.2639 , 553.8235 , ..., 200.81694, 190.63893,
        207.06682],
       [512.6017 , 472.1864 , 570.98865, ..., 184.89613, 192.95789,
        179.47856]], dtype=float32)

> file.data.dtype
Out  [3]: dtype('float32')

**Comment by ericpre** [comment]
From the description it is not clear what is the issue, for example can you clarify what you mean with "does not scale well" or "does not stretch the pixel histogram well".
If you change the dtype, you may need to normalise your data to the supported range of values (you can use [numpy.iinfo](https://numpy.org/doc/2.1/reference/generated/numpy.iinfo.html) for figure out what it is) before changing the dtype to avoid clipping values. It can be using minimum, maximum, difference of the two, or using scikit-image function, such as [rescale_intensity](https://scikit-image.org/docs/0.24.x/api/skimage.exposure.html#skimage.exposure.rescale_intensity).
